docs/en/reference/working-with-associations.rst

```diff
@@ -444,34 +427,34 @@ following code:
     $em->persist($myFirstComment); // required, if `cascade: persist` isn't set
     $em->flush();
 
-Even if you *persist* a new User that contains our new Comment this
-code requires an explicit call to
-``EntityManager#persist($myFirstComment)``. Doctrine 2 does not
-cascade the persist operation to all nested entities that are new
-as well.
-
-More complicated is the deletion of all user's comments when the user is
-removed from the system:
+Note that the Comment entity is instantiated right here in the controller.
+To avoid this, ``cascade: persist`` allows you to "hide" the Comment entity from the controller,
+only accessing it through the User entity:
 
 .. code-block:: php
 
     <?php
-    $user = $em->find('User', $deleteUserId);
-    
-    foreach ($user->getAuthoredComments() as $comment) {
-        $em->remove($comment);
-    }
-    $em->remove($user);
-    $em->flush();
+    // User entity
+    class User
+    {
+        private $id;
+        private $comments;
 
-Without the loop over all the authored comments Doctrine would use
-an UPDATE statement only to set the foreign key to NULL and only
-the User would be deleted from the database during the
-flush()-Operation.
+        public function __construct()
+        {
+            $this->id = User::new();
+            $this->comments = new ArrayCollection();
+        }
 
-To have Doctrine handle both cases automatically we can change the
-``User#commentsAuthored`` property to cascade both the "persist"
-and the "remove" operation.
+        public function comment(string $text, DateTimeInterface $time) : void
+        {
+            $this->comments->add(Comment::create($text, $time));
```

Comment:
I would add the $comment->setUser() here otherwise I can already see them copy pasting this.